ecs_model_deployer/src/index.ts

```diff
@@ -88,56 +80,34 @@ export const handler = async (event: any) => {
 
 
     const stackName = `${config.deploymentName}-${modelConfig.modelId}`;
-    if ( event.action === ACTION_DEPLOY ) {
-        const deploy_promise: Promise<ChildProcess | undefined> = new Promise( (resolve) => {
-            const cp = spawn('./node_modules/aws-cdk/bin/cdk', ['deploy', stackName, '-o', '/tmp/cdk.out']);
-
-            cp.on('close', (code) => {
-                console.log(`cdk deploy exited early, code ${code}`);
-                resolve(cp);
-            });
-
-            cp.stdout.on('data', (data) => {
-                console.log(`Got data: ${data}`);
-            });
-
-            cp.stderr.on('data', (data) => {
-                console.log(`Got err data: ${data}`);
-            });
-
-            setTimeout(() => {
-                console.log('180 second timeout');
-                resolve(undefined);
-            }, 180 * 1000);
+    const deploy_promise: Promise<ChildProcess | undefined> = new Promise( (resolve) => {
+        const cp = spawn('./node_modules/aws-cdk/bin/cdk', ['deploy', stackName, '-o', '/tmp/cdk.out']);
```

Comment:
I know we're using cdk here, but is it also possible to set a Role to be associated with the cloudformation stack? (if it doesn't do that already?)

DeleteModel doesn't have the Delete* permissions that would be required, but that's ok if the stack has an associated role that already has those permissions.

Reply:
To be honest I'm not I understand the question. I will say for the cdk function, the lambda has 'sts:AssumeRole' on 'cdk*' roles, and it assumes that role before deploying anything.

I also used to have the 'action' parameter that would let us call `cdk destroy` instead of `aws cloudformation delete-stack` directly, so if we run into permissions issues we could move back to that and have cdk just assume the right role automatically. The big drawback of this is that you would need to pass in the entire modelConfig again in order to synth

Reply:
the equivalent of: can we set the `--role-arn` property on the stack through cdk? https://docs.aws.amazon.com/cli/latest/reference/cloudformation/create-stack.html

Reply:
Looks like it defaults to cdk-cfn-exec-role... but that we can potentially change it. What would we want to change it to? If we give the Delete lambda 'sts:AssumeRole' on cdk* would that be enough?

Reply:
Gotcha, the stack will have a role arn of something CDK created, so I think all we need then is cloudformation:DeleteStack on the deletemodel lambda role